pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -119,6 +128,15 @@ func (c *StaticResourceController) WithIgnoreNotFoundOnCreate() *StaticResourceC
 	return c
 }
 
+// WithRequirement adds a requirement function, which blocks the sync method from being executed.
+// The precondition parameter should follow the convention described in the StaticResourcesRequirementFuncType.
+//
+// When the requirement is not met, the controller reports degraded status.
+func (c *StaticResourceController) WithRequirement(requirement StaticResourcesRequirementFuncType) *StaticResourceController {
+	c.requirement = requirement
```

Comment:
would have expected a slice and appending here.

Reply:
I believe this is fine. The method signature is `WithRequirement` (singular). This means we have only one requirement here. If that was plural, than I would agree with you - it should be a slice.

Reply:
I disagree. `c.WithRequirement(x).WithRequirement(y)`. That's a common pattern.